File: task-03/Ruby/Sub-task 4/sub-task-4.rb
num = File.read('input.txt')
n = num.chomp.to_i

File.open('output.txt','w') do |pattern|

	for i in 0..n-1
    		for j in i..n-1
        		pattern.print "  "
    		end	
    		for j in 0..i
        		pattern.print "* "
    		end	
    		for j in 0..i-1
        		pattern.print "* "
    		end
    		pattern.puts(" ")
	end
	for i in 0..n
    		for j in 0..i-1
        		pattern.print "  "
    		end
    		for j in i..n-1
        		pattern.print "* "
    		end
    		for j in i..n
        		pattern.print "* "
    		end
    		pattern.puts(" ")
	end
end

File: task-03/Go/Sub-task 4/sub-task-4.go
package main
import(
    "fmt"
    "io/ioutil"
    "strconv"
    "strings"

)

func main() {
    input, err := ioutil.ReadFile("input.txt")
    if err != nil {
        fmt.Println("Failed to read input file: %s", err)
    }


    n, err := strconv.Atoi(strings.TrimSpace(string(input)))
    if err != nil {
        fmt.Println("Failed to convert input to a number: %s", err)
    }


    var pattern string
    for i:=0;i<n;i++{
		for j:=i;j<n;j++{
			pattern+="  "
		}
		for j:=0;j<=i;j++{
			pattern+="* "
		}
		for j:=0;j<i;j++{
			pattern+="* "
		}
	pattern+="\n"
	}
	for i:=0;i<=n;i++{
                for j:=0;j<i;j++{
                        pattern+="  "
                }
                for j:=i;j<n;j++{
                        pattern+="* "
                }
                for j:=i;j<=n;j++{
                        pattern+="* "
                }
        pattern+="\n"
        }

    err = ioutil.WriteFile("output.txt",[]byte(pattern), 0644)
    if err != nil {
        fmt.Println("Failed to write to output file: %s", err)
    }
}
